src/NHibernate/Type/AbstractStringType.cs

```diff
@@ -58,13 +58,16 @@ public override void Set(DbCommand cmd, object value, int index, ISessionImpleme
 		{
 			var parameter = cmd.Parameters[index];
 
+			if (value is Enum)
+				value = Enum.GetName(value.GetType(), value);
+
 			//Allow the driver to adjust the parameter for the value
 			session.Factory.ConnectionProvider.Driver.AdjustParameterForValue(parameter, SqlType, value);
 
 			// set the parameter value before the size check, since ODBC changes the size automatically
 			parameter.Value = value;
 
-			if (parameter.Size > 0 && ((string)value).Length > parameter.Size)
+			if (parameter.Size > 0 && value != null && ((string)value).Length > parameter.Size)
```

Comment:
`Set` is called by `NullSafeSet` which checks for `null` and does not call `Set` in case of a `null` value. So this null check here is moot, there is no need for it.